linetools/lists/linelist.py

```diff
@@ -577,11 +577,14 @@ def all_transitions(self, line):
                     ie = row['ion']  # ionization estate
                     Ej = row['Ej']  # Energy of lower level
                     break
+            cond = []
             if Z is not None:
                 tbl = self.__getitem__((Z, ie))
                 # Make sure the lower energy level is the same too
-                cond = tbl['Ej'] == Ej
+                #cond = tbl['Ej'] == Ej
+                cond = np.array([name1.split(' ')[0] == line for name1 in tbl['name']])
                 tbl = tbl[cond]
+                tbl.sort(['Ej','wrest'])
                 # For hydrogen/deuterium this contains deuterium/hydrogen;
                 # so let's get rid of them
                 if (line == 'HI') or (line == 'DI'):
```

Comment:
I guess that with the new method based on name only, this Deuterium/Hydrogen thing is not longer needed... right?